src/app/machines/views/MachineList/MachineListControls/MachineListControls.tsx

```diff
@@ -27,42 +39,138 @@ const MachineListControls = ({
   setHiddenGroups,
   hiddenColumns,
   setHiddenColumns,
+  setSidePanelContent,
 }: MachineListControlsProps): JSX.Element => {
   const [searchText, setSearchText] = useState(filter);
+  const hasSelection = useHasSelection();
+  const sendAnalytics = useSendAnalytics();
+  const dispatch = useDispatch();
+  const [tagsSeen, setTagsSeen] = useStorageState(
+    localStorage,
+    "machineViewTagsSeen",
+    false
+  );
 
   useEffect(() => {
     // If the filters change then update the search input text.
     setSearchText(filter);
   }, [filter]);
 
+  const getTitle = useCallback(
+    (action: NodeActions) => {
+      if (action === NodeActions.TAG) {
+        const title = getNodeActionTitle(action);
+        if (!tagsSeen) {
+          return (
+            <>
+              {title} <i className="p-text--small">(NEW)</i>
+            </>
+          );
+        }
+        return title;
+      }
+      return null;
+    },
+    [tagsSeen]
+  );
+
   return (
     <Row className="machine-list-controls">
-      <Col size={3}>
-        <MachinesFilterAccordion
-          searchText={searchText}
-          setSearchText={(searchText) => {
-            setFilter(searchText);
-          }}
-        />
-      </Col>
-      <Col size={4}>
-        <DebounceSearchBox
-          onDebounced={(debouncedText) => setFilter(debouncedText)}
-          searchText={searchText}
-          setSearchText={setSearchText}
-        />
-      </Col>
-      <Col size={3}>
-        <div className="u-flex--align-baseline">
-          <div className="u-flex--grow">
-            <GroupSelect
-              grouping={grouping}
-              setGrouping={setGrouping}
-              setHiddenGroups={setHiddenGroups}
+      {!hasSelection ? (
+        <>
+          <Col size={2}>
+            <MachinesFilterAccordion
+              searchText={searchText}
+              setSearchText={(searchText) => {
+                setFilter(searchText);
+              }}
             />
-          </div>
-        </div>
-      </Col>
+          </Col>
+          <Col size={5}>
+            <DebounceSearchBox
+              onDebounced={(debouncedText) => setFilter(debouncedText)}
+              searchText={searchText}
+              setSearchText={setSearchText}
+            />
+          </Col>
+          <Col size={3}>
+            <div className="u-flex--align-baseline">
+              <div className="u-flex--grow">
+                <GroupSelect
+                  grouping={grouping}
+                  setGrouping={setGrouping}
+                  setHiddenGroups={setHiddenGroups}
+                />
+              </div>
+            </div>
+          </Col>
+        </>
+      ) : (
+        <>
+          <Col size={8}>
+            <NodeActionMenuGroup
+              alwaysShowLifecycle
+              excludeActions={[NodeActions.IMPORT_IMAGES]}
+              getTitle={getTitle}
+              hasSelection={hasSelection}
+              key="machine-list-action-menu"
```

Comment:
Is there a reason why you've set the react key manually here? Keys are useful in dynamic lists as they help React identify which items have changed. In practice this usually means an array (especially in which the order of items may change). This is not the case here.

 You have multiple elements with the same key now which results in the following warning: `Encountered two children with the same key, `machine-list-action-menu``

Reply:
Huh, I thought I'd changed this one here. I was basically following the implementation of the normal Node Action Menu, which also has a key there since it (was) rendered out in a list by the section header. 

Thought I'd changed this to be "machine-list-action-menu-group", but since we're no longer mapping them in a list and just displaying them as children directly, I'd be happy to just remove the keys here

Reply:
Yes please!